app/controllers/api/teachers_controller.rb

```diff
@@ -11,7 +11,7 @@ def create
       @teacher = Teacher.new(teacher_params)
        
       if @teacher.save
-        redirect_to @teacher
+        render json: @teacher
       else
         render 'new'
```

Comment:
instead of render you would want to throw an error response!
like:  error_response(@teacher)